sdk/core/System.ClientModel/src/Message/PipelineRequest.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace System.ClientModel.Primitives;
+
+public abstract class PipelineRequest : IDisposable
+{
+    /// <summary>
+    /// Gets or sets the request HTTP method.
+    /// </summary>
+    public string Method
+    {
+        get => GetMethodCore();
+        set => SetMethodCore(value);
+    }
+
+    protected abstract string GetMethodCore();
+
+    protected abstract void SetMethodCore(string method);
+
+    public Uri Uri
```

Comment:
Interesting pattern.   Why not just make the property abstract?   Unless I'm missing something, this would achieve the same result in a clearer form, wouldn't it?

```csharp
public abstract Uri Uri { get; set; };
```

Are we intentionally trying to discourage mocking the property for some reason?

Reply:
Unfortunately, we can't make it abstract because we newslot it in the Azure.Core derived type and that means that types deriving from Azure.Core's Request wouldn't be able to provide an implementation.  It doesn't compile!  FWIW, none of us love this pattern but we haven't been able to come up with a better one yet.  It follows the Template Pattern in FDG (chapter 9 if you're interested!)